src/server_manager/model/digitalocean.ts

```diff
@@ -15,6 +15,8 @@
 import * as location from "./location";
 import {ManagedServer} from "./server";
 
+export const DO_CLOUD_ID = 'do';
```

Comment:
There's no need for the model to have cloud ids. We have per-cloud code in app.ts. Every caller of addServer is cloud-specific. Whatever needs to be cloud-specific, can live in that code. For example you can define cloud ids in the app layer and pass that in addServer. addServer can map that id to cloud name and icon.

We have a view per server, created at addServer time. We can leverage that and set all the parameters we need there.

On a side note, the way we do the server views are suboptimal. Instead of using data binding, we are adding views explicitly. If we had a ui model and used binding it would be a bit cleaner, but that's a big change.

Reply:
If you think it will make things easier, perhaps we can have methods like `isDigitaloceanServer(server)` possibly based on some `server.isDigitalOcean()`, since interfaces don't exist in javascript (you'd check the presence of the method).